keps/sig-autoscaling/4951-configurable-hpa-tolerance/kep.yaml

```diff
@@ -0,0 +1,45 @@
+title: Configurable tolerance for HPA
+kep-number: 4951
+authors:
+  - "@pr00se"
+  - "@jm-franc"
+owning-sig: sig-autoscaling
+participating-sigs:
+  - sig-aaa
+  - sig-bbb
+status: provisional
+creation-date: 2024-11-05
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+see-also:
+  - "/keps/sig-autoscaling/853-configurable-hpa-scale-velocity"
+replaces:
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha
+
+# The most recent milestone for which work toward delivery of this KEP has been
+# done. This can be the current (upcoming) milestone, if it is being actively
+# worked on.
+#latest-milestone: "v1.19"
+
+# The milestone at which this feature was, or is targeted to be, at each stage.
+milestone:
+  alpha: "v1.33"
+  beta: "v1.34"
+  stable: "v1.36"
+
+# The following PRR answers are required at alpha release
+# List the feature gate name and the components for which it must be enabled
+feature-gates:
+  - name: HPAConfigurableTolerance
+    components:
+      - kube-apiserver
```

Comment:
I think this should be `kube-controller-manager`.

See https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager.

Reply:
Looking at [this PR](https://github.com/kubernetes/kubernetes/pull/90691), they used the feature gate in the api-server for things like [validation](https://github.com/kubernetes/kubernetes/pull/90691/files#diff-027285f56d21884d07c805d8d80861e051b7ae27b071873cf6fa3e5f3beb0fc9R319), so maybe we do need both components. If that's the case then we'll need to fill out the version skew section of the template as well. 🫤

Reply:
That's the information I was looking for, thank you both!

I've set this to `kube-apiserver` (validation) and `kube-controller-manager` (to gate the logic). I've updated the _version skew_ section.